### PR TITLE
[SYCLomatic] Update the error log of --query-api-mapping.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -1102,6 +1105,17 @@ int runDPCT(int argc, const char **argv) {
     if (RunResult && StopOnParseErr) {
       DumpOutputFile();
       if (RunResult == 1) {
+        if (DpctGlobalInfo::isQueryAPIMapping()) {
+          if (getDpctTermStr().find("use of undeclared identifier") !=
```
Comment (yihwang-nv) [inline clang/lib/DPCT/DPCT.cpp]: Let's use `StringRef.contains` instead

File: clang/lib/DPCT/DPCT.cpp [hunk 2]
```diff
@@ -1102,6 +1105,17 @@ int runDPCT(int argc, const char **argv) {
...
+              std::string::npos) {
+            ShowStatus(MigrationErrorAPIMappingWrongCUDAHeader);
+            return MigrationErrorAPIMappingWrongCUDAHeader;
+          } else if (getDpctTermStr().find("file not found") !=
```
Comment (yihwang-nv) [inline clang/lib/DPCT/DPCT.cpp]: likewise

File: clang/lib/DPCT/Error.cpp [hunk 3]
```diff
@@ -160,6 +160,14 @@ void ShowStatus(int Status, std::string Message) {
         "Error: This API mapping query is not available yet. You may get the "
         "API mapping by migrating a sample code of this API with the tool.";
     break;
+  case MigrationErrorAPIMappingWrongCUDAHeader:
+    StatusString = "Error: This API mapping query is not available in current "
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Error.cpp]: Error: Can not find ${API} in current CUDA header file: ${path}.  Please check the API name or switch to use different CUDA header file with option "-cuda-include-path".

File: clang/lib/DPCT/Error.cpp [hunk 4]
```diff
@@ -160,6 +160,14 @@ void ShowStatus(int Status, std::string Message) {
...
+                   "CUDA header files.";
+    break;
+  case MigrationErrorAPIMappingNoCUDAHeader:
+    StatusString = "Error: This API mapping query is not available because the "
+                   "header file is not avalible.";
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Error.cpp]: Error: Can not find ${API} in current CUDA header file: ${path}. Please specify the header file for $API with option "-extra-arg".

File: clang/lib/DPCT/Error.cpp (hunk 3 above)
Comment (tangjj11) [reply clang/lib/DPCT/Error.cpp]: fixed

File: clang/lib/DPCT/Error.cpp (hunk 4 above)
Comment (tangjj11) [reply clang/lib/DPCT/Error.cpp]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**